classpath/java/io/BufferedInputStream.java

```diff
@@ -22,7 +22,7 @@ public BufferedInputStream(InputStream in, int size) {
   }
   
   public BufferedInputStream(InputStream in) {
-    this(in, 4096);
+    this(in, 65536);
```

Comment:
Maybe, but it would take a lot to convince me that this is a better default for every possible use case (or even better on average).  For instance, if the file you're reading is small (say, 10KB), it's a complete waste of space to allocate a 64KB buffer.  Also keep in mind that, for blocking connections (which this is intended for, as discussed below), this will have to block in order to fill up the buffer.  A bigger buffer means longer blocking and therefore more latency.

If you ever run into your gigabytes-of-data use case again, you can always specify the larger buffer size by calling the other constructor.